python/whylogs/api/writer/writer.py

```diff
@@ -34,15 +63,88 @@ def check_interval(self, interval_seconds: int) -> None:
         an unacceptable interval.
         """
 
+    def _create_zip(self, files: List[str], dir: Optional[str] = None, zip_name: Optional[str] = None) -> str:
+        with tempfile.NamedTemporaryFile(suffix=".zip", prefix=zip_name, dir=dir, delete=False) as tmp_file:
+            with ZipFile(tmp_file, "w", allowZip64=True) as zip_file:
+                for file in files:
+                    zip_file.write(file, file.split(os.sep)[-1])
+                    # TODO: delete file?
+
+            tmp_file.flush()
+            tmp_file.seek(0)
+            return tmp_file.name
+
     @abstractmethod
     def write(
         self,
         file: Writable,
         dest: Optional[str] = None,
         **kwargs: Any,
-    ) -> Tuple[bool, str]:
-        pass
+    ) -> Tuple[bool, Union[str, List[Tuple[bool, str]]]]:
```

Comment:
What do we think about the optional list of (status, message) for multi-file `write()`s?